prime-router/src/main/kotlin/ReportStreamFilter.kt

```diff
@@ -0,0 +1,97 @@
+package gov.cdc.prime.router
+
+import kotlin.reflect.KProperty1
+import kotlin.reflect.full.memberProperties
+
+/**
+ * A ReportStreamFilter is the use (call) of one or more ReportStreamFilterDefinitions.
+ */
+typealias ReportStreamFilter = List<String>
+
+/**
+ * Enum of Fields in the ReportStreamFilters, below.  Used to iterate thru all the filters in ReportStreamFilters
+ */
+enum class ReportStreamFilterType(val field: String) {
+    JURISDICTIONAL_FILTER("jurisdictionalFilter"),
+    QUALITY_FILTER("qualityFilter"),
+    ROUTING_FILTER("routingFilter");
+
+    // Reflection, so that we can write a single routine to handle all types of filters.
+    val filterProperty = ReportStreamFilters::class.memberProperties.first { it.name == this.field }
+        as KProperty1<ReportStreamFilters, ReportStreamFilter?>
+    val receiverFilterProperty = Receiver::class.memberProperties.first { it.name == this.field }
+        as KProperty1<Receiver, ReportStreamFilter>
+}
+
+/**
+ * The set of filtering objects available per Organization or per Receiver
+ * (and someday, per Sender too?)
+ *
+ * Examples of FilterTypes that can be applied:
+ *  @param jurisdictionalFilter - used to limit the data received or sent by geographical region
+ *  @param qualityFilter - used to limit the data received or sent, by quality
+ *  @param routingFilter - used to limit the data received or sent, by who sent it.
+ * We allow a different set of filters per [topic]
+ */
+data class ReportStreamFilters(
+    val topic: String,
+    val jurisdictionalFilter: ReportStreamFilter?,
+    val qualityFilter: ReportStreamFilter?,
+    val routingFilter: ReportStreamFilter?,
+) {
```

Comment:
I decided to make this a data class with specific fields, rather than, say, a Map from "filterType" -> "set of Filters", because it makes the yaml representation a little cleaner.